run-tests.js

```diff
@@ -50,79 +50,11 @@ const spawn = require('child_process').spawn;
 
 const jest = path.join(__dirname, 'node_modules', '.bin', 'jest');
 
-const NRFJPROG_ARCHIVES_DIR = path.join(__dirname, 'nrfjprog');
-const NRFJPROG_HOME = path.join(NRFJPROG_ARCHIVES_DIR, 'unpacked');
-const PLATFORM_ARCHIVE_EXT = {
-    linux: 'Linux-x86_64.tar',
-    darwin: 'OSX.tar',
-};
-const PLATFORM = os.platform();
-
-function dirExists(dir) {
-    return new Promise(resolve => {
-        fs.stat(dir, err => {
-            if (err) {
-                resolve(false);
-            } else {
-                resolve(true);
-            }
-        });
-    });
-}
-
-function getTarFilePath() {
-    return new Promise((resolve, reject) => {
-        fs.readdir(NRFJPROG_ARCHIVES_DIR, (err, files) => {
-            const fileExt = PLATFORM_ARCHIVE_EXT[PLATFORM];
-            const tarFile = files.find(file => file.endsWith(fileExt));
-            if (tarFile) {
-                resolve(path.join(NRFJPROG_ARCHIVES_DIR, tarFile));
-            } else {
-                reject(new Error(`Unable to find tar file for ${PLATFORM}`));
-            }
-        });
-    });
-}
-
-function extractTarFile(filePath, outputDir) {
-    return new Promise((resolve, reject) => {
-        const extractor = tar.Extract({ path: outputDir })
-            .on('error', err => reject(err))
-            .on('end', () => resolve());
-        fs.createReadStream(filePath)
-            .on('error', err => reject(err))
-            .pipe(extractor);
-    });
-}
-
-function setupNrfjprog() {
-    if (PLATFORM === 'win32') {
-        console.log('NOTE: Automatic setup of nrfjprog is not possible on Windows. ' +
-            'If you see DLL errors, then run the installer from the nrfjprog directory.');
-        return Promise.resolve();
-    }
-    return dirExists(NRFJPROG_HOME)
-        .then(exists => {
-            if (exists) {
-                console.log(`The nrfjprog libraries already exist in ${NRFJPROG_HOME}. ` +
-                    'No setup required.');
-                return Promise.resolve();
-            }
-            return getTarFilePath()
-                .then(filePath => extractTarFile(filePath, NRFJPROG_HOME))
-                .then(() => console.log('The nrfjprog libraries have been set up in ' +
-                    `${NRFJPROG_HOME}.`));
-        });
-}
-
 function runTests() {
-    // Spawning a new process when running jest. By doing this, we can specify
-    // a custom LD_LIBRARY_PATH env variable so that the tests are able to load
+    // Spawning a new process when running jest. This is legacy: before,
+    // a custom LD_LIBRARY_PATH env variable was set so that the tests are able to load
```

Comment:
This whole file should not be needed anymore, now that everything is set up at install time. Then, in package.json, replace `node run-tests.sh` with `jest`.